### PR TITLE
DTS 21574 assignee details update

### PR DESCRIPTION
**Pull request by aksshriv1**

### CODE REVIEW

File: processors/jenkins/src/main/java/com/publicissapient/kpidashboard/jenkins/processor/JenkinsProcessorJobExecutor.java [hunk 1]
```diff
@@ -212,22 +223,45 @@ public boolean execute(JenkinsProcessor processor) {
...
+			JenkinsProcessor processor, ProcessorExecutionTraceLog processorExecutionTraceLog, int count,
+			ProjectBasicConfig proBasicConfig) {
+		Map<ObjectId, Set<Build>> buildsByJob = jenkinsClient.getBuildJobsFromServer(jenkinsServer, proBasicConfig);
 		if (MapUtils.isNotEmpty(buildsByJob)) {
 
 			int updatedJobs = addNewBuildDetails(buildsByJob, jenkinsServer, processor.getId());
 			count += updatedJobs;
 			log.info("Job updated for :{}", count);
+			if (!checkLastRun(processorExecutionTraceLog, proBasicConfig)) {
+				updateAssigneeDetailForBuild(jenkinsServer, processorExecutionTraceLog, proBasicConfig, buildsByJob);
```
Comment (Chittauri) [inline processors/jenkins/src/main/java/com/publicissapient/kpidashboard/jenkins/processor/JenkinsProcessorJobExecutor.java:235]: no need to add again...combine this function with addNewBuildDetails()

File: processors/github/src/main/java/com/publicissapient/kpidashboard/github/processor/GitHubProcessorJobExecutor.java [hunk 2]
```diff
@@ -227,7 +233,9 @@ public boolean execute(GitHubProcessor processor) {
 					}
 
 					List<MergeRequests> mergeRequestsList = gitHubClient.fetchMergeRequests(gitHubProcessorItem,
-							firstTimeRun, tool);
+							firstTimeRun, tool, proBasicConfig);
+
+					updateAssigneeForMerge(proBasicConfig, processorExecutionTraceLog, gitHubProcessorItem, mergeRequestsList);
```
Comment (hirbabar) [inline processors/github/src/main/java/com/publicissapient/kpidashboard/github/processor/GitHubProcessorJobExecutor.java:238]: remove unnecessary db call and use existing code

File: processors/github/src/main/java/com/publicissapient/kpidashboard/github/processor/GitHubProcessorJobExecutor.java [hunk 3]
```diff
@@ -272,6 +282,35 @@ public boolean execute(GitHubProcessor processor) {
...
+		if (proBasicConfig.isSaveAssigneeDetails()
+				&& !processorExecutionTraceLog.isLastEnableAssigneeToggleState()) {
+			List<CommitDetails> updateAuthor = new ArrayList<>();
+			commitDetailList.stream().forEach(commit -> {
+				CommitDetails commitDetailsData = commitsRepo.findByProcessorItemIdAndRevisionNumber(
+						gitHubProcessorItem.getId(), commit.getRevisionNumber());
+				commitDetailsData.setAuthor(commit.getAuthor());
+				updateAuthor.add(commitDetailsData);
+			});
+			commitsRepo.saveAll(updateAuthor);
```
Comment (hirbabar) [inline processors/github/src/main/java/com/publicissapient/kpidashboard/github/processor/GitHubProcessorJobExecutor.java:295]: remove this call

File: processors/gitlab/src/test/java/com/publicissapient/kpidashboard/gitlab/processor/GitLabClientTest.java [hunk 4]
```diff
@@ -91,9 +93,10 @@ public void testFetchCommits() throws Exception {
 		String serverResponse = getServerResponse("/gitlab-server/stashresponse.json");
 		String restUrl = new GitLabURIBuilder(repo, gitLabConfig, gitLabInfo).build();
 		restUrl = URLDecoder.decode(restUrl, "UTF-8");
+		projectBasicConfig.setSaveAssigneeDetails(true);
 		when(restTemplate.exchange(eq(restUrl), eq(HttpMethod.GET), ArgumentMatchers.any(HttpEntity.class), eq(String.class)))
 				.thenReturn(new ResponseEntity<String>(serverResponse, HttpStatus.OK));
-		List<CommitDetails> commits = gitLabClient.fetchAllCommits(repo, true,gitLabInfo);
+		List<CommitDetails> commits = gitLabClient.fetchAllCommits(repo,gitLabInfo,projectBasicConfig);
```
Comment (hirbabar) [inline processors/gitlab/src/test/java/com/publicissapient/kpidashboard/gitlab/processor/GitLabClientTest.java:99]: do proper formatting

File: processors/jenkins/src/main/java/com/publicissapient/kpidashboard/jenkins/processor/JenkinsProcessorJobExecutor.java [hunk 5]
```diff
@@ -127,7 +137,7 @@ public JenkinsProcessor getProcessor() {
 
 	/**
```
Comment (hirbabar) [inline processors/jenkins/src/main/java/com/publicissapient/kpidashboard/jenkins/processor/JenkinsProcessorJobExecutor.java:138]: remove unnessaory file formatiing, revert changes

File: processors/jenkins/src/main/java/com/publicissapient/kpidashboard/jenkins/processor/adapter/impl/JenkinsBuildClient.java [hunk 6]
```diff
@@ -25,6 +25,7 @@
 import java.util.Map;
 import java.util.Set;
 
+import com.publicissapient.kpidashboard.common.model.application.ProjectBasicConfig;
```
Comment (hirbabar) [inline processors/jenkins/src/main/java/com/publicissapient/kpidashboard/jenkins/processor/adapter/impl/JenkinsBuildClient.java:28]: all file re import

File: processors/jenkins/src/test/java/com/publicissapient/kpidashboard/processor/JenkinsBuildClientTests.java [hunk 7]
```diff
@@ -187,7 +189,7 @@ public void testGetJobDetailsChild() throws Exception {
 				eq(String.class))).thenReturn(
 						new ResponseEntity<>(getJson("instance_jobs_multibranch_pipeline.json"), HttpStatus.OK));
 
-		Map<ObjectId, Set<Build>> jobs = jenkinsClient.getBuildJobsFromServer(JENKINS_SAMPLE_SERVER_ONE);
+		Map<ObjectId, Set<Build>> jobs = jenkinsClient.getBuildJobsFromServer(JENKINS_SAMPLE_SERVER_ONE,projectBasicConfig);
```
Comment (hirbabar) [inline processors/jenkins/src/test/java/com/publicissapient/kpidashboard/processor/JenkinsBuildClientTests.java:192]: file formating

File: processors/teamcity/src/main/java/com/publicissapient/kpidashboard/teamcity/processor/TeamcityProcessorJobExecutor.java [hunk 8]
```diff
@@ -215,26 +228,57 @@ public boolean execute(TeamcityProcessor processor) {
 		return executionStatus;
 	}
 
+	private void updateAssigneeDetails(ProjectBasicConfig proBasicConfig, ProcessorToolConnection teamcityServer, ProcessorExecutionTraceLog processorExecutionTraceLog, Map<ObjectId, Set<Build>> buildsByJob) {
+		if (!checkLastRun(processorExecutionTraceLog, proBasicConfig)) {
+			updateAssignee(proBasicConfig, teamcityServer, processorExecutionTraceLog, buildsByJob);
+		}
```
Comment (hirbabar) [inline processors/teamcity/src/main/java/com/publicissapient/kpidashboard/teamcity/processor/TeamcityProcessorJobExecutor.java:234]: improve logic

File: processors/teamcity/src/main/java/com/publicissapient/kpidashboard/teamcity/processor/adapter/impl/DefaultTeamcityClient.java [hunk 9]
```diff
@@ -340,7 +343,7 @@ public Build getBuildDetails(String buildUrl, String instanceUrl, ProcessorToolC
 				return null;
 			}
 
-			return parseBuildDetailsResponse(resultJSON, buildUrl, teamcityServer);
+			return parseBuildDetailsResponse(resultJSON, buildUrl, teamcityServer,proBasicConfig);
 		} catch (UnsupportedEncodingException e) {
```
Comment (hirbabar) [inline processors/teamcity/src/main/java/com/publicissapient/kpidashboard/teamcity/processor/adapter/impl/DefaultTeamcityClient.java:347]: file formating

File: processors/bamboo/src/main/java/com/publicissapient/kpidashboard/bamboo/processor/BambooProcessorJobExecuter.java [hunk 10]
```diff
@@ -74,6 +78,8 @@ public class BambooProcessorJobExecuter extends ProcessorJobExecutor<BambooProce
 	private boolean executionStatus = true;
 	private int failureCount;
 	private int newBuildCount;
+	private LocalDate today = LocalDate.now();
+	private DateTimeFormatter dtf = DateTimeFormatter.ofPattern("uuuu-MM-dd");
```
Comment (hirbabar) [inline processors/bamboo/src/main/java/com/publicissapient/kpidashboard/bamboo/processor/BambooProcessorJobExecuter.java:83]: which format time?

**Pull request merged**